File: buerguer-pig/HamburgesasEIngredientes-Java-IvanRC/src/main/java/models/Ingrediente.java
package models;

import java.io.Serializable;

public class Ingrediente implements Serializable {
    private final int id;
    private final String nombre;
    private final double precio;

    public Ingrediente(int id, String nombre, double precio) {
        this.id = id;
        this.nombre = nombre;
        this.precio = precio;
    }

    public int getId() {
        return id;
    }

    public String getNombre() {
        return nombre;
    }

    public double getPrecio() {
        return precio;
    }

    @Override
    public String toString() {
        return "Ingrediente{" +
                "id=" + id +
                ", nombre='" + nombre + '\'' +
                ", precio=" + precio +
                '}';
    }
}

File: buerguer-pig/JCZhang-BurguerPigFicheros-Kotlin/src/main/kotlin/models/Ingrediente.kt
package models

import java.io.Serializable

class Ingrediente(var ID: Int = nextId(), var nombre: String, var precio: Double) : Serializable{
    constructor(): this(0,"",0.0)


    // id autonumérico
    companion object{
        var id = 1
        fun nextId(): Int{
            return id++
        }

        fun parseFromString(inputString: String): Ingrediente {
            val atributes = inputString.split("/")
            val id = atributes[0].toInt()
            val nombre = atributes[1].trim()
            val precio = atributes[2].toDouble()
            return Ingrediente(id, nombre, precio)
        }
    }
    override fun toString(): String {
        return "$ID/$nombre/$precio"
    }


}